sympy/functions/elementary/piecewise.py

```diff
@@ -797,13 +797,17 @@ def _eval_subs(self, old, new):
         # and avoiding invalid conditions that appear after a
         # True condition
         args = list(self.args)
+        args_exist = False
         for i, (e, c) in enumerate(args):
             c = c._subs(old, new)
             if c != False:
+                args_exist = True
                 e = e._subs(old, new)
             args[i] = (e, c)
             if c == True:
                 break
+        if not args_exist:
+            return Undefined
```

Comment:
I haven't learning about `_eval_subs` but is it ok to return something other than `self.func...` here?  Is it safer to do `if not args_exist: args = ((Undefined, True))`?

Reply:
Also, is this the best place to make this change?  Is this the only code path that looks like evaluate/simplify/subs/canonicalize/etc where we want `nan` instead of exception?

This code starts with `# this is strictly not necessary,` and yet now it will be necessary.

(Note this is not a well-informed opinion: if you've thought about this and this is the right place, then say so.)